SliceTracker/SliceTracker.py

```diff
@@ -2880,16 +2880,24 @@ def applyRegistration(self, progressCallback=None):
 
     coverProstateRegResult = self.registrationResults.getMostRecentApprovedCoverProstateRegistration()
     lastRigidTfm = self.registrationResults.getLastApprovedRigidTransformation()
+    lastApprovedTfm = self.registrationResults.getMostRecentApprovedTransform()
+    initialTransform = lastApprovedTfm if lastApprovedTfm else lastRigidTfm
 
     self.generateNameAndCreateRegistrationResult(self.currentIntraopVolume)
     parameterNode = slicer.vtkMRMLScriptedModuleNode()
     parameterNode.SetAttribute('FixedImageNodeID', self.currentIntraopVolume.GetID())
-    parameterNode.SetAttribute('FixedLabelNodeID', coverProstateRegResult.fixedLabel.GetID())
+
+    fixedLabel = self.volumesLogic.CreateAndAddLabelVolume(slicer.mrmlScene, self.currentIntraopVolume,
+                                                           self.currentIntraopVolume.GetName() + '-label')
+
+    self.runBRAINSResample(inputVolume=coverProstateRegResult.fixedLabel, referenceVolume=self.currentIntraopVolume,
+                           outputVolume=fixedLabel, warpTransform=initialTransform)
+
+    parameterNode.SetAttribute('FixedLabelNodeID', fixedLabel.GetID())
     parameterNode.SetAttribute('MovingImageNodeID', coverProstateRegResult.fixedVolume.GetID())
     parameterNode.SetAttribute('MovingLabelNodeID', coverProstateRegResult.fixedLabel.GetID())
     parameterNode.SetAttribute('TargetsNodeID', coverProstateRegResult.approvedTargets.GetID())
-    if lastRigidTfm:
-      parameterNode.SetAttribute('InitialTransformNodeID', lastRigidTfm.GetID())
+    parameterNode.SetAttribute('InitialTransformNodeID', initialTransform.GetID())
```

Comment:
per @che85, this code is not needed, since this parameter is not used. I was confused, since this tfm can be b-spline, and b-spline can't be used as brainsfit initializer.